### PR TITLE
Add json data storage backends.

### PR DESCRIPTION
**Pull request by pmclanahan**
Default backend is filesystem that works the way the current version does.
The other built-in backend is for the Database. Switching among them is a
setting with the dotted path to the backend class.

This also adds to our test coverage and greatly compartmentalizes and
simplifies the code in the ProductDetails class itself.

### CODE REVIEW

**Comment by pmclanahan** [comment]
@jgmize r?

**Comment by pmclanahan** [comment]
@fwenzel not sure if you're still interested in the goings on of this lib, but just in case you have an opinion I'd love to hear it :)

File: product_details/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='ProductDetailsFile',
+            fields=[
+                ('name', models.CharField(max_length=100, serialize=False, primary_key=True)),
```
Comment (jgmize) [inline product_details/migrations/0001_initial.py]: I think bumping the max_length up to ~250 would give us a little more room for future API versions without exceeding limitations of some DBs for uniqueness constraints.

**Comment by jgmize** [comment]
r+wc, nice work!

**Pull request merged**